src/containers/UserProfile/UserProfile.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from "react";
+import "./UserProfile.css";
+import db from "../../firebase";
+
+function UserProfile() {
+  // eslint-disable-next-line
+  const [edit, setEdit] = useState(false);
+
+  function handleClick() {
+    setEdit((edit) => !edit);
+  }
+
+  //needs work
+  function handleSubmit() {
+    console.log("Data Edited");
+  }
+
+  const [user, setUser] = useState([]);
+
+  const fetchUser = async () => {
+    const response = db.collection("users");
+    const data = await response.get();
+    data.docs.forEach((item) => {
+      setUser([...user, { id: item.id, ...item.data() }]);
+    });
+  };
+
+  //to fetch data from db
+  useEffect(() => {
+    fetchUser();
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  //func to render existing user data
+  function Card() {
+    return (
+      <div className="profileRender">
+        {user &&
+          user.map((singleUser) => {
```

Comment:
why are we using a map here ? the user won't be an Array I think it will be an object